src/main/scala/mesosphere/marathon/api/akkahttp/v2/TasksController.scala

```diff
@@ -193,10 +201,14 @@ class TasksController(
     * @return new deployment created as the result of scale operation
     */
   @SuppressWarnings(Array("all")) // async/await
-  def killAndScale(tasksByAppIdFuture: Future[Map[PathId, Seq[Instance]]], force: Boolean)(implicit identity: Identity): Future[DeploymentResult] = async {
-    val tasksByAppId = await(tasksByAppIdFuture)
-    val deploymentPlan = await(taskKiller.killAndScale(tasksByAppId, force))
-    DeploymentResult(deploymentPlan.id, deploymentPlan.version.toOffsetDateTime)
+  def killAndScale(toKillFuture: Future[Map[PathId, Seq[Instance]]], force: Boolean)(implicit identity: Identity): Future[Either[Rejection, DeploymentResult]] = async {
```

Comment:
I missed this on the first review. Why do we pass `toKillFuture` as a future? What is the motivation?

Reply:
Also, why are the passed? I rather inverse the call. This should call the method that defines what should be killed.